orangecontrib/spectroscopy/widgets/owhyper.py

```diff
@@ -736,6 +789,96 @@ def getAxis(self):
     def legend_items(self):
         return []
 
+    @property
+    def colorbar(self):
+        return self.master.legend.axis
+
+    @property
+    def vcolorbar(self):
+        return self.master.vect_legend.axis
+
+class VectorPlot(pg.GraphicsObject):
```

Comment:
What did you gain by redoing most of the PlotCurveItem? Are there visual elements you needed to modify `paint` for? If so, could you tell, me which?

I understand that making it a class could make the code more maintainable. I am worried about making a `paint` method. Bugs in there (and I see you were fixing them already :) are one of the surest ways to have crashes. Did you considering structuring your code so that it avoids paint? An example of what i mean is `VerticalPeakLine` in widgets/gui.

Reply:
From memory, this was originally done to make a more performant/light weight PlotCurveItem, as the original is quite slow with thicker lines (a known issue by the looks of things. This implementation is still quite slow but faster than PlotCurveItem). It also made the plotting of vectors with multiple colors much more convenient.